src/sentry/static/sentry/app/components/smartSearchBar/index.jsx

```diff
@@ -947,18 +1004,73 @@ const StyledInput = styled('input')`
   }
 `;
 
-const SidebarButton = styled(Button)`
-  & svg {
-    color: ${p => p.theme.gray2};
+const getMediaQuery = (size, type) => `
+  display: ${type};
+
+  @media (min-width: ${size}) {
+    display: ${type == 'none' ? 'block' : 'none'};
   }
-  &:hover svg {
+`;
+
+const InputButton = styled(Button)`
+  color: ${p => (p.isActive ? p.theme.blueLight : p.theme.gray2)};
+  margin-left: ${space(0.5)};
+  width: 18px;
+
+  &:hover {
     color: ${p => p.theme.gray3};
   }
-  .show-sidebar & svg {
-    color: ${p => p.theme.blueLight};
+
+  ${p =>
+    p.collapseIntoEllipsisMenu &&
+    getMediaQuery(p.theme.breakpoints[p.collapseIntoEllipsisMenu], 'none')};
+`;
+
+const SearchBuilderButton = styled(InputButton)`
+  margin-left: ${space(0.25)};
+  margin-right: ${space(0.5)};
+`;
+
+const StyledDropdownLink = styled(DropdownLink)`
+  display: none;
+
+  @media (max-width: ${p => p.theme.breakpoints[2]}) {
+    display: flex;
   }
 `;
 
+const DropdownElement = styled('a')`
```

Comment:
Can we use `MenuItem` here or would it just not be worth it?

Reply:
lemme try

Reply:
i wanna do this, but since we are passing around classes for the button styles it isn't really possible.